### DISCUSSION TITLE
What does this error message mean: Static website not supported?

### DISCUSSION THREAD

**Discussion by cellio** [datenstrom/community#679] [Ask a question] [answered]
I've been using Datenstrom Yellow on a Mac running 0S 10.12.6 (Sierra).  The self-hosted site and the build work fine.  I have a "home" area and a blog, and I use the extensions for sidebar, search, contact, and sitemap (with the Berlin theme).

I've just moved to a Mac running 11.6 (Big Sur).  When I run the server, the contact form and search load fine.  They are now showing up in the header bar, which they weren't before and which I want to fix, but there's a bigger problem I need to solve first (and maybe that will also fix this).

When I build the static site, I now get the following error output:

> $ php yellow.php build ../build/
ERROR building location '/contact/', HTTP/1.1 500 Server error: Static website not supported!
ERROR building location '/search/', HTTP/1.1 500 Server error: Static website not supported!
Building static website 100%... done
Yellow build: 74 files, 2 errors

I also tried on the shell provided with my web hosting and got the same errors. It's running some flavor of Linux; the results of `uname -a`(with my host redacted) are:

> Linux example.com 3.10.0-962.3.2.lve1.5.60.el7.x86_64 #1 SMP Fri Jul 23 07:07:00 EDT 2021 x86_64 x86_64 x86_64 GNU/Linux

(I don't know what that means; searching for "what linux am I running?" led me to this command.)

I'm at a loss on how to debug the build errors.  What should my next steps be?

**Comment by schulle4u** [comment, marked as answer]
As the error message says, static websites are not supported by contact and search extensions. The page itself theoretically could be made static, but you would need an external script to process the queries.

**Reply by cellio** [reply]
Thanks for the help.  I thought search had worked on the site built with the older version, but I've just checked and I was mistaken.  (It does work in the version that Yellow serves, for testing.)

Aren't these functions provided by search.php and contact.php?  I'm not experienced with PHP, but I see what looks like handler code there.  But I don't see any PHP in the built site, so I guess those are only to support `yellow serve`?  Is there some way to use them in the generated site too?

**Reply by schulle4u** [reply]
The built site only serves static files, e. g. HTML, CSS, JS and images. If your web server supports php, you might consider running Yellow dynamically to avoid these limitations. Its as simple as extracting one file, upload to your web host, and open the site in your browser. The serve command is not needed in this case.

**Reply by markseuffert** [reply]
You can test everything in the built-in web server, with `php yellow.php serve`, but not generate everything statically.

One of the design goals was to make small websites and it shouldn't matter if you serve them dynamically or statically. Datenstrom Yellow works both as content management system (CMS) and static site generator (SSG). When you use a CMS your website pages are dynamically generated when requested. When you use a SSG your whole static website will be generated in advance.  Makes things super fast and safe. There are some technical limitations in what you can do with static websites though, for example a static site generator can't provide online editing or full-text search.

If there's enough interest we might be able to add a limited search/contact feature for static websites. It might not be possible to do exactly the same thing, however a limited search might be better than nothing. It's on my personal wish list. Feedback is welcome.

**Reply by cellio** [reply]
As far as I can tell (I am new to PHP), my hosting provider supports PHP.  I see a bunch of PHP-related stuff in the CPanel list.  I'll see if I can figure out how to directly test it (hello-world.php).

I hadn't considered running Yellow dynamically on my web site; I figured I'd reduce the number of things that could go wrong by generating the site.  I'm new to all of this -- how would I run Yellow dynamically, in a way that makes sure it stays up?  The getting-started doc implies I just put the source in the directory the site is served out of?  I tried that, but when I visit the domain I get a [directory listing](https://test.cellio.org/), not the site, so there must be something else I need to do.  I realize I need something to invoke yellow.php; doing so directly doesn't seem to be right.  (I did the MultiPHP setup from CPanel, taking all the defaults, and it added php.ini to the directory.  I also set the PHP version to 7.3, the same one I'm running locally.  I'm not sure if there's other CPanel stuff I need to do?)

Typing `php yellow.php` on the command line on my host site produces the expected help -- meaning it can run in the shell.  I must be missing some wiring for the (non-static) web site, something that invokes Yellow correctly.

I'd love to see some sort of search support in the static site.  Is there a way to generate the search index during the build, so that the search could then use that?  (For contact, while a form is nicer, I can fall back to publishing an email address.)

**Reply by cellio** [reply]
Found it -- the .htaccess config described here: https://in.10437.lima.zone/help/server-configuration

**Reply by markseuffert** [reply]
Glad to hear you solved it. The latest troubleshooting can be found here: https://datenstrom.se/yellow/help/troubleshooting

**Reply by cellio** [reply]
Thanks for all the help.  I'm new to server setup and so I didn't immediately recognize how existing documentation applied to the problems I was running into.  Slowly I'm learning. :-)
